# Getting Started with Python 🐍

Python is a versatile, high-level programming language that is widely supported across all major operating systems.Now, let's write a simple Python program.

The following program displays `Hello, World!` on the screen.



In [1]:
print("Hello World!")

Hello World!



> **Note:**  A `Hello World!` program includes the basic syntax of a programming language and helps beginners understand the structure before getting started. That's why it is a common practice to introduce a new language using a `Hello World!` program.

## Working of the Program

Now, let's see how the above program works.

In Python, anything inside `print()` is displayed on the screen.

There are two things to note about `print()`:

- Everything we want to display on the screen is included inside the parentheses `()`.
- The text we want to print is placed within double quotes `" "`.

We can also use single quotes to print text on the screen. For example,

In [2]:
print('Hello World!')

Hello World!


is same as

In [3]:
print("Hello World!")

Hello World!


## #️⃣ Python Comments

Comments are hints that we add to our code to make it easier to understand. Python comments start with `#`. For example,


In [4]:
# print a number
print(25)

25


Here, `# print a number` is a comment.

Comments are completely ignored and not executed by code editors.

### *️⃣ Single-line Comment

We use the **hash (`#`)** symbol to write a single-line comment. For example,


In [10]:
# declare a variable
name = "Akshit"

# print name
print(name)    # Akshit

Akshit


In the above example, we have used three single-line comments:

- `# declare a variable`
- `# print name`
- `# John`

A **single-line comment** starts with `#` and extends up to the end of the line. We can also use single-line comments alongside the code:



In [11]:
print(name)    # Akshit

Akshit


## #️⃣ Multiline Comments

Unlike languages such as C++ and Java, Python doesn't have a dedicated method to write multi-line comments.

However, we can achieve the same effect by using the hash (`#`) symbol at the beginning of each line.

Let's look at an example.

In [12]:
# This is an example of a multiline comment
# created using multiple single-line commenced
# The code prints the text Hello World
print("Hello, World!")

Hello, World!


We can also use multiline strings as comments like:

In [13]:
'''This is an example 
of multiline comment'''
print("Hello, World!")

Hello, World!


### 📌 Why Use Comments?
We should use comments:

- For future references, as comments make our code readable.
- For debugging.
- For code collaboration, as comments help peer developers to understand each other's code.

## 🖇️ Python Variables
In programming, a variable is a container (storage area) to hold data. For example,


In [14]:
number=10

Here, number is a variable storing the value **10**.

### ⏩Assigning values to Variables in Python

As we can see from the above example, we use the assignment operator `=` to assign a value to a variable.

In [16]:
# assign value to name variable
name = 'Akshit'
print(name)

Akshit


#### 📌 Changing the Value of a Variable in Python


In [18]:
name = 'Akshit'
print(name)

# assigning a new value to name
name = 'Sonani'

print(name)

Akshit
Sonani


#### 📌 Example: Assigning multiple values to multiple variables

In [19]:
a, b, c = 5, 3.2, 'Hello'

print (a)  # prints 5
print (b)  # prints 3.2
print (c)  # prints Hello 

5
3.2
Hello


If we want to assign the same value to multiple variables at once, we can do this as:

In [20]:
x = y  = 'Hello'

print (x)  # prints Hello
print (y)  # prints Hello

Hello
Hello


## 📌 Python Literals

Literals are representations of fixed values in a program. They can be numbers, characters, or strings, etc. For example, `'Hello, World!'`, `12`, `23.0`, `'C'`, etc.

### 📌 Python Numeric Literals

Numeric Literals are immutable (unchangeable). Numeric literals can belong to **3** different numerical types: `Integer`, `Float`, and `Complex`.

#### 1) Integer Literals
Integer literals are numbers without decimal parts. It also consists of negative numbers. For example, `5`, `-11`, `0`, `12`, etc.
#### 2) Floating-Point Literals
Floating-point literals are numbers that contain decimal parts.

Just like integers, floating-point numbers can also be both positive and negative. For example, `2.5`, `6.76`, `0.0`, `-9.45`, etc.
#### 3) Complex Literals
Complex literals are numbers that represent complex numbers.

Here, numerals are in the form `a + bj`, where `a` is real and `b` is imaginary. For example, `6+9j`, `2+3j`.

### 📌 Python String Literals

In Python, texts wrapped inside quotation marks are called **string literals**.

```python
"This is a string."
```
We can also use single quotes to create strings.
```python
'This is also a string.'
```

## 🔁 Python Type Conversion

In programming, type conversion is the process of converting data of one type to another. For example: converting `int` data to `str`.

There are two types of type conversion in Python.

- Implicit Conversion - automatic type conversion
- Explicit Conversion - manual type conversion

### 📌 Implicit Type Conversion
In certain situations, Python automatically converts one data type to another. This is known as implicit type conversion.

#### Example 1: Converting integer to float

Let's see an example where Python promotes the conversion of the lower data type (integer) to the higher data type (float) to avoid data loss.

In [21]:
integer = 200
float = 200.15

sum = integer + float

# display new value and resulting data type

print("value: ",sum)
print("Data Type: ",type(sum))

value:  400.15
Data Type:  <class 'float'>


In the above example, we have created two variables: `integer` and `float` of `int` and `float` type respectively.

Then we added these two variables and stored the result in `sum`.

As we can see new_number has value **400.15** and is of the `float` data type.

> **Note:**
> - We get `TypeError`, if we try to add `str` and `int`. For example, `'12' + 23`. Python is not able to use Implicit Conversion in such conditions.
> - Python has a solution for these types of situations which is known as Explicit Conversion.

### 📌 Explicit Type Conversion
In Explicit Type Conversion, users convert the data type of an object to required data type.

We use the built-in functions like int(), float(), str(), etc to perform explicit type conversion.

This type of conversion is also called typecasting because the user casts (changes) the data type of the objects.

#### Example 2: Addition of string and integer Using Explicit Conversion

In [25]:
num_string = '12'
num_integer = 23

print("Data type of num_string before Type Casting:",type(num_string))

# explicit type conversion

num_string = int(num_string)

print("Data type of num_string after Type Casting:",type(num_string))

sum = num_string + num_integer

print("Sum: ",sum)
print("Data type of sum after Type Casting:",type(sum))

Data type of num_string before Type Casting: <class 'str'>
Data type of num_string after Type Casting: <class 'int'>
Sum:  35
Data type of sum after Type Casting: <class 'int'>


In the above example, we have created two variables: `num_string` and `num_integer` with `str` and `int` type values respectively. Notice the code,

In [26]:
num_string = int(num_string)

Here, we have used `int()` to perform explicit type conversion of `num_string` to integer type.

After converting `num_string` to an integer value, Python is able to add these two variables.

Finally, we got the `sum` value i.e **35** and data type to be `int`.

### 📌 Key Points to Remember
1) Type Conversion is the conversion of an object from one data type to another data type.
2) Implicit Type Conversion is automatically performed by the Python interpreter.
3) Python avoids the loss of data in Implicit Type Conversion.
4) Explicit Type Conversion is also called Type Casting, the data types of objects are converted using predefined functions by the user.
5) In Type Casting, loss of data may occur as we enforce the object to a specific data type.

## 👨🏻‍🚀 Python Operators

### 📌 Types of Python Operators

1) Arithmetic Operators
2) Assignment Operators
3) Comparison Operators
4) Logical Operators
5) Bitwise Operators
6) Special Operators


#### 1️⃣ Python Arithmetic Operators

In [1]:
sub = 10-5

Here, `-` is an arithmetic operator

| Operator |operation|
|----------|---------|
| `+`      |Addtion|
| `-`      |Substraction|
| `*`      |Multiplication|
| `/`      |Division|
| `//`     |Floor division|
| `%`      |Modulo|
| `**`     |Power|
